appium/webdriver/common/touch_action.py

```diff
@@ -126,12 +134,14 @@ def release(self):
 
         return self
 
-    def perform(self):
+    def perform(self: T) -> T:
         """Perform the action by sending the commands to the server to be operated upon
 
         Returns:
             `TouchAction`: self instance
         """
+        if self._driver is None:
+            raise AttributeError('Set driver to constructor as a argument when to create the instance.')
```

Comment:
I would rather verify if `self._driver` is not None when passing it to the constructor. Also https://stackoverflow.com/questions/8297526/proper-exception-to-raise-if-none-encountered-as-argument makes sense

Reply:
> I would rather verify if self._driver is not None when passing it to the constructor.
Some functions in `TouchAction` don't need `driver`, so its check cannot be moved to the constructor.

> Also https://stackoverflow.com/questions/8297526/proper-exception-to-raise-if-none-encountered-as-argument makes sense
Thanks.